databus/src/main/java/com/bazaarvoice/emodb/databus/core/DefaultDatabus.java

```diff
@@ -580,18 +618,25 @@ public void replay(String subscription, Date since) {
         checkState(since == null || new DateTime(since).plus(DatabusChannelConfiguration.REPLAY_TTL).isAfterNow(),
                 "Since timestamp is outside the replay TTL.");
         String source = ChannelNames.getMasterReplayChannel();
-        final Subscription destination = getSubscription(subscription);
+        final OwnedSubscription destination = getSubscriptionByName(subscription);
+        final DatabusAuthorizer.DatabusAuthorizerByOwner authorizer = _databusAuthorizer.owner(destination.getOwnerId());
 
         _eventStore.copy(source, subscription, new Predicate<ByteBuffer>() {
             @Override
-            public boolean apply(ByteBuffer eventData) {
-                return _subscriptionEvaluator.matches(destination, eventData);
+            public boolean apply(ByteBuffer eventDataBytes) {
+                try {
+                    SubscriptionEvaluator.MatchEventData eventData = _subscriptionEvaluator.getMatchEventData(eventDataBytes);
+                    return _subscriptionEvaluator.matches(destination, eventData)
+                            && authorizer.canReceiveEventsFromTable(eventData.getTable().getName());
```

Comment:
Related to me earlier comment. If we make authorization a part of SubscriptionEvaluator, then caller methods wouldn't risk exposing events without checking authorization for them.